recipes/soci/all/conanfile.py

```diff
@@ -0,0 +1,181 @@
+        if self.options.with_oracle:
+            # self.requires("oracle_db/0.0.0") # TODO add support for oracle
+            raise ConanInvalidConfiguration("{} ORACLE {} ".format(prefix, message))
+        if self.options.with_firebird:
+            # self.requires("firebird/0.0.0") # TODO add support for firebird
+            raise ConanInvalidConfiguration("{} firebird {} ".format(prefix, message))
+        if self.options.with_mysql:
+            self.requires("libmysqlclient/8.0.17")
+        if self.options.with_postgresql:
+            self.requires("libpq/11.5")
+        if self.options.with_boost:
+            self.requires("boost/1.73.0")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename(self.name + "-" + self.version, self._source_subfolder)
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+
+        self._cmake = CMake(self)
+
+        self._cmake.definitions["SOCI_SHARED"]  = self.options.shared
+        self._cmake.definitions["SOCI_TESTS"]   = False
+        self._cmake.definitions["SOCI_CXX11"]   = True
+
+        if self.options.shared:
+            self._cmake.definitions["SOCI_STATIC"] = False
+
+        self._cmake.definitions["SOCI_EMPTY"]       = self.options.empty
+        self._cmake.definitions["WITH_SQLITE3"]     = self.options.with_sqlite3
+        self._cmake.definitions["WITH_DB2"]         = self.options.with_db2
+        self._cmake.definitions["WITH_ODBC"]        = self.options.with_odbc
+        self._cmake.definitions["WITH_ORACLE"]      = self.options.with_oracle
+        self._cmake.definitions["WITH_FIREBIRD"]    = self.options.with_firebird
+        self._cmake.definitions["WITH_MYSQL"]       = self.options.with_mysql
+        self._cmake.definitions["WITH_POSTGRESQL"]  = self.options.with_postgresql
+        self._cmake.definitions["WITH_BOOST"]       = self.options.with_boost
+
+        self._cmake.configure()
+
+        return self._cmake
+
+    def build(self):
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def package(self):
+        self.copy("LICENSE_1_0.txt", dst="licenses", src=self._source_subfolder)
+        self.copy("*soci*.lib", dst="lib", src="lib", keep_path=False, symlinks=True)
+        self.copy("*soci*.so*", dst="lib", src="lib", keep_path=False, symlinks=True)
+        self.copy("*.a",        dst="lib", src="lib", keep_path=False, symlinks=True)
+        self.copy("*.dylib",    dst="lib", src="lib", keep_path=False, symlinks=True)
+        self.copy("*.dll",      dst="bin", src="bin", keep_path=False, symlinks=True)
+
+        if self._cmake:
+            self._cmake.install()
+            tools.rmdir(os.path.join(self.package_folder, "cmake"))
+            tools.rmdir(os.path.join(self.package_folder, "lib64"))
+
+    def package_info(self):
+        self.cpp_info.includedirs   = ['include']
+        self.cpp_info.libdirs       = ['lib']
```

Comment:
I noticed some weird artifacts in the package folder looking into the logs. https://c3i.jfrog.io/c3i/misc/logs/pr/3924/34/soci/4.0.1/3054ef85ecfd32282ae23a7676e9e8d634ded5b0/create_stdout.txt

```text
-- Installing: /home/conan/w/cci_PR-3924/.conan/data/soci/4.0.1/_/_/package/3054ef85ecfd32282ae23a7676e9e8d634ded5b0/include/soci/soci-config.h.in
-- Installing: /home/conan/w/cci_PR-3924/.conan/data/soci/4.0.1/_/_/package/3054ef85ecfd32282ae23a7676e9e8d634ded5b0/include/soci/soci-config.h
```

It seems you copy the libs but they are installed by cmake. No need for duplication

I'd suggest you test the following

```suggestion
    def package(self):
        self.copy("LICENSE_1_0.txt", dst="licenses", src=self._source_subfolder)
        cmake = self._configure_cmake()
        cmake.install()
        tools.rmdir(os.path.join(self.package_folder, "cmake"))
        tools.remove_files_by_mask(os.path.join(self.package_folder, "include", "soci"), "*.in*")

    def package_info(self):
        self.cpp_info.libdirs       = ['lib64']
```

Reply:
The `lib64` folder was not accepted by the hooks, so I removed it